testing/citest/tests/kube_smoke_test.py

```diff
@@ -389,6 +389,324 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
+
+  def make_delete_stage_daemonset(self, **kwargs):
+      bindings = self.bindings
+      result = {
+          'refId': '1',
+          'type': 'destroyServerGroup',
+          'cloudProvider': 'kubernetes',
+		  'cloudProviderType': 'kubernetes',
+		  'cluster': self.TEST_APP+'tstv001',
+		  'credentials': bindings['SPINNAKER_KUBERNETES_ACCOUNT'],
+          'interestingHealthProviderNames': ['KubernetesService'],
+		  'kind': 'DaemonSet',
+		  'name': 'Destroy Server Group',
+                  'target': 'current_asg_dynamic',
+		  'namespaces': ['default'],
+		  'preCondition': 'true'
+      }
+      result.update(kwargs)
+      return result
+
+  def make_delete_stage_statefulset(self, **kwargs):
+      bindings = self.bindings
+      result = {
+          'refId': '1',
+          'type': 'destroyServerGroup',
+          'cloudProvider': 'kubernetes',
+		  'cloudProviderType': 'kubernetes',
+		  'cluster': self.TEST_APP+'tstv001',
+		  'credentials': bindings['SPINNAKER_KUBERNETES_ACCOUNT'],
+          'interestingHealthProviderNames': ['KubernetesService'],
+		  'kind': 'StatefulSet',
+                  'target': 'current_asg_dynamic',
+		  'name': 'Destroy Server Group',
+		  'namespaces': ['default'],
+		  'preCondition': 'true'
+      }
+      result.update(kwargs)
+      return result
+
+  def execute_delete_daemonset_pipeline(self):
+      bindings = self.bindings
+      payload = self.agent.make_json_payload_from_kwargs(
+          job=[{
+              'type': 'manual',
+              'user': '[anonymous]'
+          }],
+          description='Delete Daemonset in Application' + self.TEST_APP,
+          application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      (builder.new_clause_builder('Daemonset Deleted', retryable_for_secs=15)
+       .get_resources(
+          'daemonsets', extra_args=[self.TEST_APP+'tstv001', '--namespace', self.TEST_NAMESPACE])
+       .contains_path_value('kind', 'DaemonSet'))
+      return st.OperationContract(
+          self.new_post_operation(
+              title='Delete_daemonset', data=payload, path='pipelines/' + self.TEST_APP + '/daemonset-delete-pipeline',
+              status_class=st.SynchronousHttpOperationStatus),
+          contract=builder.build())
+
+  def execute_delete_statefulset_pipeline(self):
+      bindings = self.bindings
+      payload = self.agent.make_json_payload_from_kwargs(
+          job=[{
+              'type': 'manual',
+              'user': '[anonymous]'
+          }],
+          description='Delete StatefulSet in Application' + self.TEST_APP,
+          application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      (builder.new_clause_builder('StatefulSet Deleted', retryable_for_secs=15)
+       .get_resources(
+          'statefulsets', extra_args=[self.TEST_APP+'tstv001', '--namespace', self.TEST_NAMESPACE])
+       .contains_path_value('kind', 'StatefulSet'))
```

Comment:
Similar to above, this contract looks supicious. If it is correct, please explain it.

Reply:
Yes you are right , i have changed the code as per already did in 'delete_server_group'.